Newspack/Newspack/Controllers/AssetsViewController.swift

```diff
@@ -123,14 +130,66 @@ extension AssetsViewController {
 extension AssetsViewController {
 
     func cellFor(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "AssetCell", for: indexPath)
-        cell.textLabel?.text = storyAsset.name + " " + String(storyAsset.order)
-        cell.detailTextLabel?.text = storyAsset.uuid.uuidString
-        cell.accessoryType = .disclosureIndicator
+        switch storyAsset.assetType {
+        case .textNote:
+            return configureTextCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .image:
+            return configurePhotoCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .video:
+            return configureVideoCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .audioNote:
+            return configureAudioCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        }
+    }
+
+    func configureTextCell(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> TextNoteTableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: TextNoteTableViewCell.reuseIdentifier, for: indexPath) as! TextNoteTableViewCell
```

Comment:
I'm selling [this extension API](https://github.com/Automattic/simplenote-ios/blob/develop/Simplenote/Classes/UITableView+Simplenote.swift#L35) REALLY REALLY cheap 😂

Reply:
Nice! I'm totally buying :D